### ISSUE TITLE
When training a pose model, how are losses calculated with regard to the different visibility of points?

### ISSUE THREAD

**Issue by dongxinyu1030** [ultralytics/ultralytics#3093]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

When training a pose model, how are losses calculated with regard to the different visibility of points? If non visible, will loss be calculated with respect to that point? 
When loading labels, it seems visibility states are ignored.

**Comment by thanhhuyenpt** [comment]
> I believe visibility loss is handled by BCE and regression loss only computed for visible points.

Hi @glenn-jocher, I train yolov8-pose with 5 classes (4 classes with no keypoints (for detection only) and 1 class has keypoints). Is it feasible to modify loss for each classes? For example, I just want to calculate detection loss for 4 classes which have no keypoint, and calculate pose loss for class which has keypoint, because when I finish my training, the average of pose result is so low because of the effect of the classes with no keypoints. 
Thanks in advance!

**Comment by glenn-jocher** [comment]
@thanhhuyenpt hello,

To achieve what you're asking for - calculating different loss types depending on the class - you would need to modify the loss computation code in the model implementation. Specifically, you would need to split your batch into data entries pertaining to the different classes. Then, for each class, apply your desired loss function whether it be detection loss for your 4 classes and pose loss for the class with keypoints. But please note that this operation might be computationally more expensive due to operations being calculated per class instead of on the whole batch. 

Regarding the low average pose results, it could also be a factor of imbalance between the classes or differing amounts data of data available for each class. If there is significantly less data for the class with keypoints, their contribution to the overall pose loss could be drowned out by detection loss for the other classes. In such a case, you might want to consider over-sampling the classes with keypoints or applying class weights during loss computation to mitigate this issue.

I hope this helps, and feel free to report back if you experience any difficulties.
